distributed/scheduler.py

```diff
@@ -7735,6 +7735,14 @@ def check_idle(self) -> int | None:
             self.idle_since = time()
             return self.idle_since
 
+        if self.jupyter:
+            last_activity = (
```

Comment:
BTW I don't think either of these code coverage warnings are accurate. If I put a `raise RuntimeError` at either of those lines, the new tests in `test_jupyter.py` fail.